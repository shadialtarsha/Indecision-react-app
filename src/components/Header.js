import React from 'react';
import { connect } from 'react-redux';
import { startLogout } from '../actions/auth';

export const Header = props => (
  <div className="header">
    <div className="container header__container">
      <div>
        <h1 className="header__title">{props.title}</h1>
        {props.subtitle && <h2 className="header__subtitle">{props.subtitle}</h2>}
      </div>
      <button className="button button--link" onClick={props.startLogout}>
        Logout
      </button>
    </div>
  </div>
);
Header.defaultProps = {
  title: 'Indecision',
};

const mapDispatchToProps = dispatch => ({
  startLogout: () => dispatch(startLogout()),
});

export default connect(undefined, mapDispatchToProps)(Header);
