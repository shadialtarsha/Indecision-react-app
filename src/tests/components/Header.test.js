import React from 'react';
import { shallow } from 'enzyme';
import Header from '../../components/Header';

test('should render Header correctly', () => {
  const wrapper = shallow(<Header title="new Title" subtitle="new subtitle" />);
  expect(wrapper).toMatchSnapshot();
});

test('should render Header with default title correctly', () => {
  const wrapper = shallow(<Header subtitle="new subtitle" />);
  expect(wrapper).toMatchSnapshot();
});
