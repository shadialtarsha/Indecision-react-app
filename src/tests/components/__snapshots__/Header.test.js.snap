// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header correctly 1`] = `
<div
  className="header"
>
  <div
    className="container"
  >
    <h1
      className="header__title"
    >
      new Title
    </h1>
    <h2
      className="header__subtitle"
    >
      new subtitle
    </h2>
  </div>
</div>
`;

exports[`should render Header with default title correctly 1`] = `
<div
  className="header"
>
  <div
    className="container"
  >
    <h1
      className="header__title"
    >
      Indecision
    </h1>
    <h2
      className="header__subtitle"
    >
      new subtitle
    </h2>
  </div>
</div>
`;
